### PR TITLE
Reinforce Asteroid Walls + Porous Rock Type

### PR DESCRIPTION
**Pull request by Kurfursten**
🆑 
* tweak: Porous rock is the new name for rock that contains air pockets. It has a minimum mining time (slow picks are not affected but fast picks are hindered).
* rscadd: Asteroid rock may now be reinforced with metal. Doing so destroys the minerals inside. You can weld the reinforced wall to remove the minerals. Porous rock can also be reinforced (does not destroy air pockets).

### CODE REVIEW

**Comment by brndd** [comment]
I think this desperately needs spritework. What I had in mind was a tile where the edges are metal plates, and the interior of the tile is roidrock. If spriters are feeling productive, a variant with wooden reinforcements would be nice for flavour. Coding-wise those would be made using planks instead of metal, and function much like plank floors in that screwdrivers detach them cleanly and crowbars detach them destructively.

Excellent job coding-wise though, I think this will solve the problem of exterior Roidstation maintenance being too easy to breach nicely.

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -190,6 +220,27 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
...
+			S.use(2)
+			user.visible_message("<span class='notice'>[user] finishes installing reinforcements to \the [src].</span>", \
+				"<span class='notice'>You finish installing reinforcements to \the [src].</span>")
+			var/old_type = type
+			var/old_name = name
+			var/turf/simulated/wall/X = ChangeTurf(/turf/simulated/wall/r_rock)
+			if(X)
+				X.name = "reinforced [old_name]"
+				X.dismantle_type = old_type
+				X.add_hiddenprint(user)
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: Doesn't `add_fingerprint`  one line below also do what this does?

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -190,6 +220,27 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
...
 		return
 
+	if(istype(W,/obj/item/stack/sheet/metal))
+		var/obj/item/stack/sheet/S = W
+		if(S.amount < 2)
+			return
+		user.visible_message("<span class='notice'>[user] starts installing reinforcements to \the [src].</span>", \
+			"<span class='notice'>You start installing reinforcements to \the [src].</span>")
+		if(do_after(user, src, 40))
+			if(S.amount < 2)
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: you can check the return value of use instead

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -190,6 +220,27 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
...
+				return
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: an error message would be nice

File: code/modules/mining/mine_turfs.dm [hunk 4]
```diff
@@ -190,6 +220,27 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
 		to_chat(user, "<span class='notice'>[bicon(P)] [src] has been excavated to a depth of [2*excavation_level]cm.</span>")
 		return
 
+	if(istype(W,/obj/item/stack/sheet/metal))
+		var/obj/item/stack/sheet/S = W
+		if(S.amount < 2)
+			return
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: an error message would be nice

File: code/modules/mining/mine_turfs.dm [hunk 5]
```diff
@@ -57,11 +59,39 @@
...
 
+/turf/simulated/wall/r_rock
+	name = "reinforced rock"
+	desc = "It has metal struts that need to be welded away before it can be mined."
+	icon_state = "porousbar"
+	dismantle_type = /turf/unsimulated/mineral
+	girder_type = null
+	canSmoothWith = "/turf/unsimulated/wall/r_rock=0&/turf/unsimulated/mineral=0"
+
+/turf/simulated/wall/r_rock/New()
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: is there a reason you're not calling `..()`?

File: code/modules/mining/mine_turfs.dm [hunk 6]
```diff
@@ -57,11 +59,39 @@
 	no_finds = 1
 	mined_type = /turf/simulated/floor/asteroid
 
+/turf/simulated/wall/r_rock
+	name = "reinforced rock"
+	desc = "It has metal struts that need to be welded away before it can be mined."
+	icon_state = "porousbar"
+	dismantle_type = /turf/unsimulated/mineral
+	girder_type = null
+	canSmoothWith = "/turf/unsimulated/wall/r_rock=0&/turf/unsimulated/mineral=0"
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: needs to be a list as of #19231

File: code/modules/mining/mine_turfs.dm [hunk 7]
```diff
@@ -10,6 +10,7 @@
 	opacity = 1
 	density = 1
 	blocks_air = 1
+	canSmoothWith = "/turf/unsimulated/wall/r_rock=0&/turf/unsimulated/mineral=0"
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: needs to be a list as of #19231

File: code/modules/mining/mine_turfs.dm [hunk 8]
```diff
@@ -190,6 +220,27 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
 		to_chat(user, "<span class='notice'>[bicon(P)] [src] has been excavated to a depth of [2*excavation_level]cm.</span>")
 		return
 
+	if(istype(W,/obj/item/stack/sheet/metal))
+		var/obj/item/stack/sheet/S = W
+		if(S.amount < 2)
+			return
+		user.visible_message("<span class='notice'>[user] starts installing reinforcements to \the [src].</span>", \
+			"<span class='notice'>You start installing reinforcements to \the [src].</span>")
+		if(do_after(user, src, 40))
```
Comment (DamianX) [inline code/modules/mining/mine_turfs.dm]: `40` -> `4 SECONDS`

**Comment by boy2mantwicethefam** [comment]
ATOMIC

**Comment by SonixApache** [comment]
so porous rock is just simulated mineral rock right

**Comment by SonixApache** [comment]
Spawn "/obj/structure/girder/wood

**Comment by SonixApache** [comment]
so porous rock is just simulated mineral rock right

**Comment by Kurfursten** [comment]
Porous rock isn't new, it's just a new name for something that already existed on Roidstation. It's technically unsimmed.

**Comment by SonixApache** [comment]
please ignore my 5 deleted comments

**Pull request merged**